S-L-Murphey/Gamer-Rater-Client
# Added ability to see all games, game details, and create new game

#### Changes Made


1. Added file `ApplicationViews`, 'CategoryProvider`, `GameForm`, and `GameDetails` to allow user to see games, game details, and create new games.

#### Steps to Review

1. Checkout to this branch locally
   ```
   git fetch --all
   git checkout slm-gamerrater-client
   ```
2. Ensure the backend and client are both running.
4. Test app functionality.
   
   > Example: When user clicks on games, they should see 2 games. When they click on one of the games, they should see that game's details. When they click Register New Game, they should be able to enter all relevant information for that new game and when click Create Game, they should go back to the games page with that new info.
   
5. View code file.
   > Confirm file modifications are present as indicated above.
   > Confirm no unused code or extraneous comments exist.
